### ISSUE TITLE
tugger-apple-codesign fails to build, preventing pyoxidizer installation

### ISSUE THREAD

**Issue by jkwill87** [indygreg/PyOxidizer#411]
PyOxidizer seems like a cool project and I'm looking forward to trying out. Unfortunately, I'm running into a bit of a problem. I'm following the getting started guide, running `cargo install pyoxidizer` as instructed, but the command fails as tugger-apple-codesign fails to build:
```
   ...
   Compiling reqwest v0.11.3
   Compiling tugger-common v0.4.0
   Compiling cryptographic-message-syntax v0.3.0
   Compiling tugger-windows v0.5.0
   Compiling tugger-snapcraft v0.5.0
   Compiling tugger-rust-toolchain v0.3.0
   Compiling tugger-windows-codesign v0.3.0
   Compiling tugger-apple-codesign v0.4.0
   Compiling starlark v0.3.1
error[E0282]: type annotations needed
  --> /Users/jkwill87/.cargo/registry/src/github.com-1ecc6299db9ec823/tugger-apple-codesign-0.4.0/src/macos.rs:66:18
   |
66 |     search.limit(i32::MAX as _);
   |                  ^^^^^^^^^^^^^ cannot infer type
   |
   = note: type must be known at this point

error: aborting due to previous error

For more information about this error, try `rustc --explain E0282`.
error: could not compile `tugger-apple-codesign`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `pyoxidizer v0.16.0`, intermediate artifacts can be found at `/var/folders/zq/rd7bfx9d0bgbpy49vfhpht9h0000gn/T/cargo-installDkXrRx`

Caused by:
  build failed
```

[Here is a gist of the whole verbose log output](https://gist.github.com/jkwill87/27838eb3cdfc0153279042dd8c62937e). 

Normally I wouldn't post an issue for a failing dependency but it seems to be part of this project. I attempted to build it against rustc stable (stable-x86_64-apple-darwin 1.52.1 9bc8c42bb 2021-05-09) and nightly (1.54.0-nightly 0a8629bff 2021-06-11).

Any guidance would be appreciated. Thanks!

**Comment by Alyetama** [comment]
I have the same issue.  I retired with a fresh rust and cargo installation, but I still see the same error. 

```
default host triple: aarch64-apple-darwin
default toolchain: stable (default)
profile: default

rustc 1.53.0 (53cb7b09b 2021-06-17)
```

Although this appears during installation, I can still use it.  However, when I run my project, it throws this error:

```
<jemalloc>: Unsupported system page size
<jemalloc>: Unsupported system page size
memory allocation of 4 bytes failed
error: cargo run failed
```

**Comment by indygreg** [comment]
The compilation error will be fixed via a push in a few minutes.

> Although this appears during installation, I can still use it. However, when I run my project, it throws this error:
>
> ```
> <jemalloc>: Unsupported system page size
> <jemalloc>: Unsupported system page size
> memory allocation of 4 bytes failed
> error: cargo run failed
> ```

This seems to be a different error. Could you please file a new issue @Alyetama?

**Comment by hwooson12** [comment]
I have the same issue too. Is it currently fixed?

```
$ rustc --version
rustc 1.54.0 (a178d0322 2021-07-26)
```

```
$ cargo install pyoxidizer
    Updating crates.io index
  Installing pyoxidizer v0.16.0
   Compiling libc v0.2.98
   Compiling proc-macro2 v1.0.28
   Compiling unicode-xid v0.2.2
   Compiling syn v1.0.74
   Compiling cfg-if v1.0.0
   Compiling autocfg v1.0.1
   Compiling typenum v1.13.0
   ...
   Compiling cryptographic-message-syntax v0.3.0
   Compiling tugger-apple-codesign v0.4.0
error[E0282]: type annotations needed
  --> /Users/hwooson/.cargo/registry/src/github.com-1ecc6299db9ec823/tugger-apple-codesign-0.4.0/src/macos.rs:66:18
   |
66 |     search.limit(i32::MAX as _);
   |                  ^^^^^^^^^^^^^ cannot infer type
   |
   = note: type must be known at this point

error: aborting due to previous error

For more information about this error, try `rustc --explain E0282`.
error: could not compile `tugger-apple-codesign`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `pyoxidizer v0.16.0`, intermediate artifacts can be found at `/var/folders/7y/pfr0rhl122vgg_nqrrkkk55h0000gp/T/cargo-install3hdzQv`

Caused by:
  build failed
```

**Comment by indygreg** [comment]
The (just-released) PyOxidizer 0.17 release should fix this bug.
